src/rgw/rgw_civetweb_frontend.cc

```diff
@@ -64,6 +64,9 @@ int RGWCivetWebFrontend::run()
     conf_map["run_as_user"] = std::move(uid_string);
   }
 
+  /* Set authentication domains, this is needed for handling absolute uris in
+     requests */
+  conf_map["authentication_domain"] = g_conf->rgw_dns_name;
```

Comment:
@theanalyst this is not necessarily the only domain that we support. We'd have a different domain for static websites, and nowadays we check every domain to match buckets (see code in rgw_rest.cc). Instead of having civetweb limit us to a specific domain, I'd rather have it send us everything and we can make the decision in rgw.